lib/worker/src/web3storage.rs

```diff
@@ -57,144 +23,80 @@ pub enum Error {
     SerdeJSONError(#[source] serde_json::Error, String),
 }
 
-pub const DEFAULT_BASE_URL: &str = "https://api.web3.storage";
+#[async_trait]
+pub trait Web3StorageClient: Clone + Send {
+    async fn upload(
+        &self,
+        stream: impl StreamExt<Item = Result<Bytes, google_cloud_storage::http::Error>>
+            + Sync
+            + Send
+            + 'static,
+        filename: &str,
+    ) -> Result<UploadResponse, Error>;
+}
 
 #[derive(Clone)]
-pub struct Web3StorageClient {
+pub struct Web3StorageImpl {
     base_url: String,
-    token: String,
 }
 
-impl Web3StorageClient {
-    pub fn new(base_url: String, token: String) -> Self {
-        Self { base_url, token }
+impl Web3StorageImpl {
+    pub fn new(base_url: String) -> Self {
+        Self { base_url }
     }
+}
+
+#[async_trait]
+impl Web3StorageClient for Web3StorageImpl {
```

Comment:
changes the web3 storage client to make a call to the service. It returns a `UploadResponse` object

Reply:
sounds good!